.github/workflows/python-integration-test.yml

```diff
@@ -19,29 +19,52 @@ on:
 jobs:
   test:
     name: Integration Tests
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, linux, ARM64]
     strategy:
       fail-fast: false
       matrix:
         python-version: ["3.9"]
 
     steps:
-      - name: Checkout
+      - name: Checkout (PR)
         if: ${{ github.event_name  == 'pull_request'}}
         uses: actions/checkout@v3
         with:
           ref: ${{ github.head_ref }}
-      - name: Checkout
+
+      - name: Checkout (Push)
         if: ${{ github.event_name  == 'push'}}
         uses: actions/checkout@v3
         with:
           ref: ${{ github.ref_name }}
+
       - name: Install Dependencies
         run: |
-          python -m pip install --upgrade pip
           python -m venv venv_test
           source venv_test/bin/activate
+          python -m pip install --upgrade pip
           python -m pip install -r requirements_test.txt
+
+      - name: Add EdgePi GPG Key
+        run: |
+          python -m venv venv_test
+          source venv_test/bin/activate
+          wget -qO - https://deb.edgepi.com/edgepi-release.gpg | sudo gpg --dearmor -o /usr/share/keyrings/edgepi-release.gpg
+
+      - name: Setup EdgePi APT Repository
+        run: |
+          sudo tee /etc/apt/sources.list.d/edgepi.list >/dev/null <<'EOF'
+          deb [signed-by=/usr/share/keyrings/edgepi-release.gpg] https://deb.edgepi.com/main bullseye main
+          deb [signed-by=/usr/share/keyrings/edgepi-release.gpg] https://deb.edgepi.com/test bullseye test
+          deb [signed-by=/usr/share/keyrings/edgepi-release.gpg] https://deb.edgepi.com/nightly bullseye nightly
+          EOF
+
+      - name: Update APT and Install EdgePi RPC Server
+        run: |
+          sudo apt-get update
+          apt-cache pkgnames | grep '^edge'
+          sudo apt-get install edgepi-rpc-server
```

Comment:
remove these steps from the workflow and add them to the readme.md file under intg test folder. mention that these are the requirements for the intg. tests to run.